stream/src/main/scala/org/apache/pekko/stream/impl/fusing/FlatMapPrefix.scala

```diff
@@ -43,8 +43,10 @@ import pekko.util.OptionVal
         .mandatoryAttribute[Attributes.NestedMaterializationCancellationPolicy]
         .propagateToNestedMaterialization
     val matPromise = Promise[M]()
-    val logic = new GraphStageLogic(shape) with InHandler with OutHandler {
-      val accumulated = collection.mutable.Buffer.empty[In]
+    object logic extends GraphStageLogic(shape) with InHandler with OutHandler {
+      private var left = if (n < 0) 0 else n
```

Comment:
we already have a `require(n >= 0)` above, I don't think we need to add code to deal with the `n < 0` case here.